### ISSUE TITLE
Jetson decoder(nvv4l2dec, and NVMPI) stops sending video packets in kodi, if streams are split, or cut to commercial.

### ISSUE THREAD

**Issue by GavinDarkglider** [xbmc/xbmc#21022]
## Bug report
When watching split Streams with commercialsm when the video crashes decoder hangs with double dequee, and audio from next video continues. Tested with both nvv4l2dec and nvmpi_dec for jetson boards.

Here is a clear and concise description of what the problem is:
Attached is the ffmpeg log from the stream for both NVMPI decoder and NVV4L2 decoder
and this was tested in my libreelec 10 build for the Nintendo Switch with south park addon.

## Expected Behavior
When commercial end in stream video/audio of next stream should both continue, not just audio, with hanged video.

## Actual Behavior
Video freezes, and audio from next part of stream continues.

## Possible Fix
No clue, or I would be submitting patches.

Logs for both decoders are attached.

### To Reproduce

1. Use nvmpi or nvv4l2dec decoder, you will have to patch ffmpeg yourself, if you need patches, ask, and I will give them, I would post them now, but I want them to go upstream to their respective projects first. Anyway, after you have a properly patched ffmpeg, play any stream with split commecials, like early seasons of south park from the addon, and issue stated above happens.

### Debuglog
[kodi.log](https://github.com/xbmc/xbmc/files/8111415/kodi.log)
[kodi_debug.log](https://github.com/xbmc/xbmc/files/8112497/kodi.log)
[nvv4l2dec_kodi_log.txt](https://github.com/xbmc/xbmc/files/8111098/nvv4l2dec_kodi_log.txt)
[nvmpi_kodi_log.txt](https://github.com/xbmc/xbmc/files/8111101/nvmpi_kodi_log.txt)

### Screenshots 
Here are some links or screenshots to help explain the problem:
I cant/dont know how to take screenshots in libreelec. That being said, there are builds linked below for anyone with a nintendo switch to test.

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

End goal would be to have a decent media center build for Jetson Boards/Nintendo switch using libreelec, or kodi in ubuntu.

### Your Environment
Used Operating system:

Libreelec, built from this lakka tree: https://github.com/GavinDarkglider/Lakka-LibreELEC/commits/fix-nvmpi. Latest decoder nvv4l2 decoder stuff is not included, but nvmpi stuff should build automatically and mostly work(Seeking in video is broken in nvmpim and will wait as long as you seek either direction before video will continue, this is fixed in latest nvv4l2 stuff I am waiting on to be pushed here: https://gitlab.com/switchroot/switch-l4t-multimedia/FFmpeg/-/commits/4.4-l4t When this has been pushed, I will update the patches in Lakka tree)

 - Operating system version/name: LibreELEC Custom for Nintendo Switch
 - Kodi version: 19.3-440f47e475dd8a48e0a6d41349e83b74890f3fbe8275d3e401d3c50f5b9ea09b

Here is a link to the LibreELEC build, with proper nvv4l2 and nvmpi decoder, due to decoder fixing, I split libavcodec from the builds and it needs to go in libreelec/storage/libavcodec.so

(NVV4L2 Build)
https://nightly.builds.lakka.tv/members/gavin/nvv4l2-ctc-moonlight-fix/LibreELEC-Switch.aarch64-10.0-devel-20220218013737-594ff67.tar
https://nightly.builds.lakka.tv/members/gavin/nvv4l2-ctc-moonlight-fix/libavcodec.so

(NVMPI Build)
https://nightly.builds.lakka.tv/members/gavin/nvmpi-logging/LibreELEC-Switch.aarch64-10.0-devel-20220221174300-594ff67.tar
https://nightly.builds.lakka.tv/members/gavin/nvmpi-logging/libavcodec.so

**Comment by lrusak** [comment]
You haven't posted Kodi logs so it's not really clear how this is related to Kodi.

**Comment by GavinDarkglider** [comment]
> You haven't posted Kodi logs so it's not really clear how this is related to Kodi.
Hold on, I will dump them from the terminal...... Basically, I was more focused on ffmpeg side of it, than the kodi side of it.....
[Kodi_log_nvmpi_full.txt](https://github.com/xbmc/xbmc/files/8111301/Kodi_log_nvmpi_full.txt)

here is the full log, from kodi start, to the point I ended the screen, if this isnt what you are looking for, let me know how to get what you need.

**Comment by lrusak** [comment]
> > You haven't posted Kodi logs so it's not really clear how this is related to Kodi.
> > Hold on, I will dump them from the terminal...... Basically, I was more focused on ffmpeg side of it, than the kodi side of it.....
> > [Kodi_log_nvmpi_full.txt](https://github.com/xbmc/xbmc/files/8111301/Kodi_log_nvmpi_full.txt)
> 
> here is the full log, from kodi start, to the point I ended the stream, if this isnt what you are looking for, let me know how to get what you need. Also, ignore the Pulse symlink stuff, that is an issue in Libreelec, when using fat32 storage.

The Kodi log on LibreELEC is located at `/storage/.kodi/temp/kodi.log`

**Comment by GavinDarkglider** [comment]
> > > You haven't posted Kodi logs so it's not really clear how this is related to Kodi.
> > > Hold on, I will dump them from the terminal...... Basically, I was more focused on ffmpeg side of it, than the kodi side of it.....
> > > [Kodi_log_nvmpi_full.txt](https://github.com/xbmc/xbmc/files/8111301/Kodi_log_nvmpi_full.txt)
> > 
> > 
> > here is the full log, from kodi start, to the point I ended the stream, if this isnt what you are looking for, let me know how to get what you need. Also, ignore the Pulse symlink stuff, that is an issue in Libreelec, when using fat32 storage.
> 
> The Kodi log on LibreELEC is located at `/storage/.kodi/temp/kodi.log`

[kodi.log](https://github.com/xbmc/xbmc/files/8111415/kodi.log)

I am not seeing anything useful in that log, maybe you will.

**Comment by lrusak** [comment]
> > > > You haven't posted Kodi logs so it's not really clear how this is related to Kodi.
> > > > Hold on, I will dump them from the terminal...... Basically, I was more focused on ffmpeg side of it, than the kodi side of it.....
> > > > [Kodi_log_nvmpi_full.txt](https://github.com/xbmc/xbmc/files/8111301/Kodi_log_nvmpi_full.txt)
> > > 
> > > 
> > > here is the full log, from kodi start, to the point I ended the stream, if this isnt what you are looking for, let me know how to get what you need. Also, ignore the Pulse symlink stuff, that is an issue in Libreelec, when using fat32 storage.
> > 
> > 
> > The Kodi log on LibreELEC is located at `/storage/.kodi/temp/kodi.log`
> 
> [kodi.log](https://github.com/xbmc/xbmc/files/8111415/kodi.log)
> 
> I am not seeing anything useful in that log, maybe you will. Added that log to initial post as well.

You'll need to enable debugging.
https://kodi.wiki/view/Settings/System/Logging#Enable_debug_logging

**Comment by GavinDarkglider** [comment]
[kodi.log](https://github.com/xbmc/xbmc/files/8112497/kodi.log)
Debug log of issue.

**Comment by matthuisman** [comment]
If the streams are using x-discontinuity for ad breaks (like most do), then ffmpeg doesn't yet support x-discontinuity.

**Comment by GavinDarkglider** [comment]
Well, I am not going to rewrite addons to use inputstream, also doesnt inputstream adaptive mainly handle widevine, which isnt supported on arm64?

**Comment by matthuisman** [comment]
Well then you'll need to rewrite ffmpeg to handle it...

**Comment by GavinDarkglider** [comment]
This might help, if kodi calls ffmpeg directly, but not if it uses libavcodec. http://ffmpeg.org/pipermail/ffmpeg-devel/attachments/20180316/739542e4/attachment.obj

I guess this could just be ported into kodi as well.

**Comment by matthuisman** [comment]
pretty sure the kodi devs will have no interest in integrating a 4 year old patch.
There must be a reason it hasn't been merged into ffmpeg...

**Comment by GavinDarkglider** [comment]
I was saying it could lead to ideas, not that it would be 1:1 compatible.

**Comment by matthuisman** [comment]
"No clue, or I would be submitting patches."

You now have a clue - so get to work :)
